subprojects/architecture-test/src/test/java/org/gradle/architecture/test/InternalNullabilityTest.java

```diff
@@ -0,0 +1,664 @@
+        "org.gradle.api.internal.jvm",
+        "org.gradle.api.internal.lambdas",
+        "org.gradle.api.internal.notations",
+        "org.gradle.api.internal.plugins",
+        "org.gradle.api.internal.project.ant",
+        "org.gradle.api.internal.project.antbuilder",
+        "org.gradle.api.internal.properties",
+        "org.gradle.api.internal.provider.sources",
+        "org.gradle.api.internal.provider.sources.process",
+        "org.gradle.api.internal.provider.support",
+        "org.gradle.api.internal.resolve",
+        "org.gradle.api.internal.rules",
+        "org.gradle.api.internal.runtimeshaded",
+        "org.gradle.api.internal.specs",
+        "org.gradle.api.internal.tasks.compile.daemon",
+        "org.gradle.api.internal.tasks.compile.filter",
+        "org.gradle.api.internal.tasks.compile.incremental",
+        "org.gradle.api.internal.tasks.compile.incremental.analyzer",
+        "org.gradle.api.internal.tasks.compile.incremental.asm",
+        "org.gradle.api.internal.tasks.compile.incremental.cache",
+        "org.gradle.api.internal.tasks.compile.incremental.classpath",
+        "org.gradle.api.internal.tasks.compile.incremental.compilerapi",
+        "org.gradle.api.internal.tasks.compile.incremental.compilerapi.constants",
+        "org.gradle.api.internal.tasks.compile.incremental.compilerapi.deps",
+        "org.gradle.api.internal.tasks.compile.incremental.deps",
+        "org.gradle.api.internal.tasks.compile.incremental.processing",
+        "org.gradle.api.internal.tasks.compile.incremental.recomp",
+        "org.gradle.api.internal.tasks.compile.incremental.serialization",
+        "org.gradle.api.internal.tasks.compile.incremental.transaction",
+        "org.gradle.api.internal.tasks.compile.processing",
+        "org.gradle.api.internal.tasks.compile.reflect",
+        "org.gradle.api.internal.tasks.compile.tooling",
+        "org.gradle.api.internal.tasks.execution.statistics",
+        "org.gradle.api.internal.tasks.options",
+        "org.gradle.api.internal.tasks.scala",
+        "org.gradle.api.internal.tasks.testing",
+        "org.gradle.api.internal.tasks.testing.detection",
+        "org.gradle.api.internal.tasks.testing.filter",
+        "org.gradle.api.internal.tasks.testing.junit",
+        "org.gradle.api.internal.tasks.testing.junit.result",
+        "org.gradle.api.internal.tasks.testing.junitplatform",
+        "org.gradle.api.internal.tasks.testing.logging",
+        "org.gradle.api.internal.tasks.testing.operations",
+        "org.gradle.api.internal.tasks.testing.processors",
+        "org.gradle.api.internal.tasks.testing.report",
+        "org.gradle.api.internal.tasks.testing.results",
+        "org.gradle.api.internal.tasks.testing.testng",
+        "org.gradle.api.internal.tasks.testing.worker",
+        "org.gradle.api.internal.tasks.userinput",
+        "org.gradle.api.java.archives.internal",
+        "org.gradle.api.plugins.antlr",
```

Comment:
:question: Shouldn't this be included in the public api?

I found these packages that I expected be public api:

- org.gradle.api.plugins.antlr
- org.gradle.api.plugins.scala
- org.gradle.api.publish.ivy.plugins
- org.gradle.api.publish.ivy.tasks
- org.gradle.api.publish.maven.plugins
- org.gradle.api.publish.maven.tasks
- org.gradle.api.publish.plugins
- org.gradle.api.publish.tasks
- org.gradle.api.reflect
- org.gradle.api.specs
- org.gradle.api.tasks.testing
- org.gradle.api.tasks.testing.testng
- org.gradle.api.tasks.wrapper
- org.gradle.buildinit.plugins
- org.gradle.ide.visualstudio.plugins
- org.gradle.ide.visualstudio.tasks
- org.gradle.ide.xcode.plugins
- org.gradle.ide.xcode.tasks
- org.gradle.jvm
- org.gradle.jvm.tasks
- org.gradle.language.assembler.plugins
- org.gradle.language.assembler.tasks
- org.gradle.plugin.devel.plugins
- org.gradle.plugins.ear
- org.gradle.plugins.ide.api
- org.gradle.plugins.ide.eclipse
- org.gradle.plugins.ide.eclipse.model
- org.gradle.plugins.ide.idea
- org.gradle.plugins.ide.idea.model
- org.gradle.plugins.signing
- org.gradle.testing.base
- org.gradle.testing.base.plugins
- org.gradle.testing.jacoco.tasks
- org.gradle.util

I tried to ignore old software model packages but might have failed for a couple.

Either the set of internal api in arch tests is wrong or we missed adding packages to the public api over time :roll_eyes:

Reply:
The internal API selected for this test is effectively a "not public API" as it was used by the arch tests. I'll look into why so many API packages ended up being in the "internal" API.

Reply:
The issue turned out to be that the class selectors for the test caught too much in the net.

I adjusted the test to work with *packages* that are not public API, instead of classes.